migration/orm_test.go

```diff
@@ -0,0 +1,154 @@
+package migration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestSchemaVersionedBucket(t *testing.T) {
+	const thisPkgName = "testpkg"
+
+	reg := newRegister()
+
+	reg.MustRegister(1, &MyModel{}, NoModification)
+	reg.MustRegister(2, &MyModel{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		msg := m.(*MyModel)
+		msg.Cnt += 2
+		return msg.err
+	})
+	reg.MustRegister(3, &MyModel{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		panic("not implemented")
+	})
+
+	db := store.MemStore()
+
+	schema := NewSchemaBucket()
```

Comment:
okay, so this is the global bucket that stores all schema info

Reply:
Yes.
It is a bucket like any others. It holds `Schema` entries and no special logic. It is a basic key-value store for package -> schema version storage.

Register on the other hand does not know anything about package versions. It can apply migration only if you tell it how far to migrate https://github.com/iov-one/weave/blob/c4117093fbbdd9e1207ed2d66b2db1f5eb9a6c9a/migration/register.go#L94

Those two are merged in custom bucket and handler implementations - there the schema version for the package is taken out of the database and register is told to apply migrations up to current schema version  https://github.com/iov-one/weave/blob/c4117093fbbdd9e1207ed2d66b2db1f5eb9a6c9a/migration/orm.go#L48-L64